server/src/test/java/org/elasticsearch/script/ScriptStatsTests.java

```diff
@@ -46,7 +49,17 @@ public void testXContent() throws IOException {
             "      {\n" +
             "        \"context\" : \"contextB\",\n" +
             "        \"compilations\" : 100,\n" +
+            "        \"compilations_history\" : {\n" +
```

Comment:
Could you please add some unit tests for serialization of TimeSeries? There is logic in there, for example, for not emitting when empty, and also ensuring some parts could be zero. Also tests for stream serialization should include pre 8.0 behavior.

Reply:
Added.